esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -345,25 +345,28 @@ def visualize_and_save_performance(performance: 'xr.DataArray', cfg: dict,
     log_provenance(caption, filename_data, cfg, ancestors)
 
 
-def compute_overall_mean(dataset):
+def compute_overall_mean(dataset, contributions):
     """Normalize all variables in a dataset and return their mean."""
     if 'perfect_model_ensemble' in dataset.dims:
         median_dim = ['perfect_model_ensemble', 'model_ensemble']
     else:
         median_dim = 'model_ensemble'
     normalized = dataset / dataset.median(dim=median_dim)
 
+    weights = xr.DataArray([contributions[vg] for vg in dataset],
```

Comment:
I know you're using 'contributions' to avoid confusion with the 'weights' in the remainder of this script, but in this line you're going from one term to the other anyway. So I would suggest to just use 'weights' inside the scope of this function, and then outside it you can still use 'contributions'.

Reply:
done!